### ISSUE TITLE
Slowness in orders - orders when all stores are selected in Multistore

### ISSUE THREAD

**Issue by ocio87** [PrestaShop/PrestaShop#27232]
### Prerequisites

- [X] I understand and accept the project's [code of conduct](https://github.com/PrestaShop/PrestaShop/blob/develop/CODE_OF_CONDUCT.md)
- [X] I have already [searched in existing issues](https://github.com/PrestaShop/PrestaShop/issues?q=is%3Aissue+label%3ABug) and found no previous report of this bug.

### Describe the bug and add screenshots

Hello,

I have just configured the multistore but it has been identified that when it is selected with all the stores and Orders - Orders is entered, it is slow to load the page (11-12 Sec). If a specific store is selected, the order list load is normal (1.5 to 3 sec).

[https://drive.google.com/file/d/1-esjJvlC-l-_PdFw_ewGxFJj_PQ68xwC/view?usp=sharing](url)

### Expected behavior

The loading behavior should be similar to when a single store is selected. The new store has only created one order. It shouldn't influence the loading time as much.

### Steps to reproduce

Select all stores in Multi-store.
Enter orders - orders
High wait time for page load.

### PrestaShop version(s) where the bug happened

1.7.8.2

### PHP version(s) where the bug happened

7.4

**Comment by Hlavtox** [comment]
Can you try to enable debug profilling in cofing/defines.inc.php? It will display all statistics at the end of each page and you can see exactly the timings of each DB query. That could help debug it.

**Comment by hibatallahAouadni** [comment]
Hello @ocio87 

Could you please follow @Hlavtox instructions and make a screenshot.
Please check and feedback.

PS: thanks @Hlavtox for your quick reply 🚀 

Thanks!

**Comment by ocio87** [comment]
Hello,

Query with all stores:

![image](https://user-images.githubusercontent.com/12537826/148399450-993d515c-29b7-4cf1-9b71-4bd9db6ff768.png)

> SELECT 
  CONCAT(
    cu.`firstname`, ' ', cu.`lastname`
  ) AS `customer`, 
  o.id_order, 
  o.reference, 
  o.total_paid_tax_incl, 
  os.paid, 
  osl.name AS osname, 
  o.id_currency, 
  cur.iso_code, 
  o.current_state, 
  o.id_customer, 
  cu.`id_customer` IS NULL as `deleted_customer`, 
  os.color, 
  o.payment, 
  s.name AS shop_name, 
  o.date_add, 
  cu.company, 
  cl.name AS country_name, 
  o.invoice_number, 
  o.delivery_number, 
  (
    SELECT 
      IF(
        count(so.id_order) > 0, 
        0, 
        1
      ) 
    FROM 
      ps_orders so 
    WHERE 
      (so.id_customer = o.id_customer) 
      AND (so.id_order < o.id_order) 
    LIMIT 
      1
  ) AS new 
FROM 
  ps_orders o 
  LEFT JOIN ps_customer cu ON o.id_customer = cu.id_customer 
  LEFT JOIN ps_currency cur ON o.id_currency = cur.id_currency 
  INNER JOIN ps_address a ON o.id_address_delivery = a.id_address 
  LEFT JOIN ps_order_state os ON o.current_state = os.id_order_state 
  LEFT JOIN ps_shop s ON o.id_shop = s.id_shop 
  INNER JOIN ps_country c ON a.id_country = c.id_country 
  INNER JOIN ps_country_lang cl ON c.id_country = cl.id_country 
  AND cl.id_lang = : context_lang_id 
  LEFT JOIN ps_order_state_lang osl ON os.id_order_state = osl.id_order_state 
  AND osl.id_lang = : context_lang_id 
WHERE 
  o.`id_shop` IN (: context_shop_ids) 
ORDER BY 
  o.id_order desc 
LIMIT 
  50

Query with a store:

![image](https://user-images.githubusercontent.com/12537826/148400516-d391f504-ee02-4de0-8473-fe6fd6b4cde8.png)

**Comment by Hlavtox** [comment]
@ocio87 Do you think you can try this?
- Open: `\src\Core\Grid\Query\OrderQueryBuilder.php` and find `public function getSearchQueryBuilder(SearchCriteriaInterface $searchCriteria)`
- Comment out `$this->addNewCustomerField($qb);` and add this above to the select? `->addSelect('1 as new')`

**Comment by ocio87** [comment]
You are the best, with that change the consultation improved.

![image](https://user-images.githubusercontent.com/12537826/148403786-d48e31cb-018c-4977-9afd-45b087cfd0da.png)

**Comment by Hlavtox** [comment]
So, now we must investigate, how to get the "new client" information faster :D

**Comment by ocio87** [comment]
You're right, after the change the client column everything remained in Yes.

![image](https://user-images.githubusercontent.com/12537826/148409217-8750c45f-bcde-4b19-8cfa-6bb887984800.png)

**Comment by Hlavtox** [comment]
@ocio87 Try one more thing.

- Go back to `\src\Core\Grid\Query\OrderQueryBuilder.php` and put it back to what it was.
- Find `$qb->addSelect('(' . $this->getNewCustomerSubSelect() . ') AS new');` and change it to `$qb->addSelect('IF((SELECT so.id_order FROM ps_orders so WHERE (so.id_customer = o.id_customer) AND (so.id_order < o.id_order) LIMIT 1) IS NOT NULL, 0, 1) AS new');`

We will see if the COUNT is or the subselect itself is the performace issue 👍

**Comment by ocio87** [comment]
It Works very well.

![image](https://user-images.githubusercontent.com/12537826/148410145-92dbed4b-a590-4a45-bb23-3cad7f524b8a.png)

![image](https://user-images.githubusercontent.com/12537826/148410306-a780f708-95eb-4ca6-9dfa-6febc139ae16.png)

**Comment by hibatallahAouadni** [comment]
Thanks @Hlavtox for all you replies :muscle: waiting for your PR :rocket: 
@ocio87 did you encounter this issue with older version?

PS: Related to this PR #26221

**Comment by ocio87** [comment]
Hello @hibatallahAouadni ,

I don't know, I'm just starting to enable multistore to the site that has been in place since version 1.5.x.

He hadn't had the need to enable.

**Comment by hibatallahAouadni** [comment]
Ping @PrestaShop/prestashop-core-developers could someone please verified this issue :pray: 
Thanks in advance!

**Comment by Hlavtox** [comment]
@atomiix You worked on the last PR. Do you think you could implement my solution properly?

**Comment by ocio87** [comment]
Is it okay to keep doing this solution manually in later versions while @atomiix gives us an answer?

Thanks!

**Comment by davidcylke** [comment]
@Hlavtox you're the man, again! I would pay for your performance improvements files mods :D

**Comment by Hlavtox** [comment]
@davidcylke HAHA 😄❤️

Guys, we should make a PR for this if it works. @ocio87

**Comment by ocio87** [comment]
Hello @Hlavtox ,

I see that they did not apply this improvement to version 8.0.1.

![image](https://user-images.githubusercontent.com/12537826/223906972-6adedcd3-3497-4d68-b4ae-33084f9ae29c.png)

Slowness is no longer present in this version?

**Comment by bpodwinski** [comment]
> [@ocio87](https://github.com/ocio87) Try one more thing.
> 
> * Go back to `\src\Core\Grid\Query\OrderQueryBuilder.php` and put it back to what it was.
> * Find `$qb->addSelect('(' . $this->getNewCustomerSubSelect() . ') AS new');` and change it to `$qb->addSelect('IF((SELECT so.id_order FROM ps_orders so WHERE (so.id_customer = o.id_customer) AND (so.id_order < o.id_order) LIMIT 1) IS NOT NULL, 0, 1) AS new');`
> 
> We will see if the COUNT is or the subselect itself is the performace issue 👍

Thank you, this fixed my problem! With 18,000 orders, it took 43 seconds to load the page on the latest version of Prestashop 8.2.

**Comment by bpodwinski** [comment]
[I explain the problem here with a fix](https://github.com/PrestaShop/PrestaShop/pull/26221#issuecomment-2476167372)

**Comment by BenTen** [comment]
@bpodwinski 

You should propose a new PR instead than write in a already Merged PR

**Comment by boo-code** [comment]
Measured on `9.2.x`, MySQL 8.4, a shop with 60,039 orders, Sell > Orders page 1 with the default sort:

```
paginated list query   0.14 s
count query            0.13 s
```

The cost this report describes came from the display query: `ORDER BY o.id_order DESC` forced a
temporary table and a filesort, so the 20-plus display columns and the correlated "new customer"
sub-select were evaluated for **every** matching order before the `LIMIT`. #41932, merged into `9.2.x`
on 2026-08-31, resolves the paginated page of order ids first and hydrates the display columns for
those rows only. On the same fixture that query went from 5.9 s to 0.13 s.

On that code the multistore join costs nothing measurable: dropping `LEFT JOIN ps_shop s` leaves the
count query at 0.13 s, and the plan for `o.id_shop IN (1)` is identical to
`o.id_shop IN (1,2,3,4,5,6,7,8,9)` - `ps_orders` carries an `id_shop` index and the id query sorts on
the primary key either way.

`8.2.x` is security-only, so #41932 is not backported there.

The fixture above is 60,039 orders on one shop, not 140,000 across nine, so it bounds what these
numbers show: the filesort that produced the reported seconds is gone, measured at that size.
